### ISSUE TITLE
undefinedConnection name for a connection definition 

### ISSUE THREAD

**Issue by excitement-engineer** [graphql/graphql-relay-js#151]
In my code I have created a connection as follows: 

```js
const { connectionType: JokeItemConnection } = connectionDefinitions({
  nodeType: new GraphQLNonNull(GraphQLJoke)
});

export default new GraphQLObjectType({
  name: "User",
  description: "Represents a user of the joke API.",
  fields: () => ({
    //... other fields have been omitted
    jokes: {
      type: new GraphQLNonNull(JokeItemConnection),
      description: "All the user's jokes.",
      args: connectionArgs,
      resolve: (user: User, args, { viewer, loaders }: Context) =>
        connectionFromPromisedArray(
          user.jokes(loaders.joke),
          args
        )
    }
  })
});
```

Next, when I look in graphiql I see that the name of the connection is `undefinedConnection`:

<img width="486" alt="screen shot 2017-02-25 at 14 52 05" src="https://cloud.githubusercontent.com/assets/19558321/23331668/9b54d55a-fb6a-11e6-850c-34dafd0ed29e.png">

When I now change the connection definition as follows: 

```js
const { connectionType: JokeItemConnection } = connectionDefinitions({
  nodeType: GraphQLJoke
});
```

Then the name of the connection is correct!

Is this a bug?

**Comment by sibelius** [comment]
@excitement-engineer is this still a problem for you?

**Comment by excitement-engineer** [comment]
Hey @sibelius! Thanks for reminding me:) I created a PR that fixes this issue, see #177.

**Comment by IvanGoncharov** [comment]
fixed in #331
